### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by Gh-Reza**
## Changes:

- Created necessary variables for Post and User actions
- Added the basic view for: index and show actions of PostController and UserContrroller

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,7 @@
-Root view
\ No newline at end of file
+<ul>
+  <% @users.each do |user| %>
+    <h4><%= user.id %></h4>
+    <li><h1><%= user.name %></h1></li>
+    <h4>Number of posts: <%= user.posts.count %></h4>
+  <% end %>
+</ul>
```
Comment (Qoosim) [inline app/views/users/index.html.erb]: - [ ] Great job creating views for your app. However, there're few things to do before you could move on to the next milestone.
- [ ] It is required that you index view page should match the [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_root_users.png?token=GHSAT0AAAAAACJF5B2AX2L63CTAV4ZLMVOGZKD6ETQ). :point_down: 

username

- After creating two users and starting your app, this is how your view page displayed. :point_down: 

your-page

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,7 @@
-Root view
\ No newline at end of file
+<ul>
+  <% @users.each do |user| %>
+    <h4><%= user.id %></h4>
+    <li><h1><%= user.name %></h1></li>
```
Comment (Qoosim) [inline app/views/users/index.html.erb]: - [ ] It is also required that clicking on each user take us to the user show page. Your user is not clickable. Kindly fix that.

### Expected show page

show-page

### Your Page
[your-app.webm](https://github.com/Gh-Reza/BlogApp/assets/26597432/7b248587-7add-40f6-b94f-659181835153)

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,20 @@
...
+    <ul>
+      <% @posts.each do |post| %>
+        <li>
+          <h4>Post: #<%= post.id %></h4>
+          <p><%= post.text %></p>
+        </li>
+      <% end %>
+    </ul>
+  </div>
+</div>
```
Comment (Qoosim) [inline app/views/users/show.html.erb]: - [ ] User show page is required to a button that has `See All Posts` text that opens  user's post index page. That is yet to be implemented. Kindly fix that. It should look something similar to this. :point_down: 

```
<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>
``` 

### Expected Page
all-posts

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Gh-Reza, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @Gh-Reza, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Added click events ✔️ 
- Views match wireframe 👍 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**